test/performance/connection/http_transport_export_test.py

```diff
@@ -0,0 +1,232 @@
+import csv
+from pathlib import Path
+from typing import Callable
+
+import pandas as pd
+import polars as pl
+import pytest
+from pyarrow import dataset
+
+from pyexasol import ExaConnection
+
+
+def get_csv_length(filepath: Path) -> int:
+    with Path(filepath).open(mode="r", encoding="utf-8") as file:
+        reader = csv.reader(file)
+        return len(list(reader))
+
+
+def get_list_length(exported_list: list) -> int:
+    return len(exported_list)
+
+
+def get_pandas_dataframe_length(exported_pandas: pd.DataFrame) -> int:
+    return exported_pandas.shape[0]
+
+
+def get_polars_dataframe_length(exported_polars: pl.DataFrame) -> int:
+    return exported_polars.shape[0]
+
+
+@pytest.mark.parametrize(
+    "export_method, file_length_function",
+    [
+        pytest.param("export_to_list", get_list_length, id="list"),
+        pytest.param("export_to_pandas", get_pandas_dataframe_length, id="pandas"),
+        pytest.param("export_to_polars", get_polars_dataframe_length, id="polars"),
+    ],
+)
+def test_export_methods_to_memory(
+    benchmark,
+    benchmark_specs,
+    tmp_source_directory,
+    connection: ExaConnection,
+    fill_sales_table,
+    export_method: str,
+    file_length_function: Callable,
+):
+    def func_to_be_measured():
+        return getattr(connection, export_method)(
+            query_or_table=fill_sales_table,
+            query_params={"columns": ["SALES_TIMESTAMP", "PRICE", "CUSTOMER_NAME"]},
+        )
+
+    result = benchmark.pedantic(
+        func_to_be_measured, iterations=1, rounds=benchmark_specs.rounds
+    )
+
+    number_rows = file_length_function(result)
+
+    assert number_rows == benchmark_specs.final_data_size
+
+
+@pytest.mark.parametrize(
+    "export_method, file_extension, file_length_function",
+    [
+        pytest.param("export_to_file", "csv", get_csv_length, id="file"),
+    ],
+)
+def test_export_methods_to_file(
+    benchmark,
+    benchmark_specs,
+    tmp_source_directory,
+    connection: ExaConnection,
+    fill_sales_table,
+    export_method: str,
+    file_extension: str,
+    file_length_function: Callable,
+):
+    export_dst = tmp_source_directory / f"test_data.{file_extension}"
+
+    def func_to_be_measured():
+        return getattr(connection, export_method)(
+            dst=export_dst,
+            query_or_table=fill_sales_table,
+            query_params={"columns": ["SALES_TIMESTAMP", "PRICE", "CUSTOMER_NAME"]},
+        )
+
+    benchmark.pedantic(
+        func_to_be_measured,
+        iterations=1,
+        rounds=benchmark_specs.rounds,
+        warmup_rounds=benchmark_specs.warm_up_rounds,
+    )
+
+    number_rows = file_length_function(export_dst)
+    assert number_rows == benchmark_specs.final_data_size
+
+
+def test_export_to_parquet(
```

Comment:
A few different variants here; at least on the performance side, there weren't significant deviations in times. We'd likely only retain this test for the export_to_parquet.